python/ray/ml/tests/test_api.py

```diff
@@ -53,6 +54,31 @@ def test_scaling_config():
     DummyTrainer(scaling_config=None)
 
 
+def test_scaling_config_validate_config():
```

Comment:
Nit: it might be beneficial to split up this test into three tests.

From [Microsoft's unit testing best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-acts):

> When writing your tests, try to only include one Act per test. Common approaches to using only one act include:
> * Create a separate test for each act.
> * Use parameterized tests.

> Why?
> * When the test fails it is not clear which Act is failing.
> * Ensures the test is focussed on just a single case.
> * Gives you the entire picture as to why your tests are failing.

> Multiple Acts need to be individually Asserted and it is not guaranteed that all of the Asserts will be executed. In most unit testing frameworks, once an Assert fails in a unit test, the proceeding tests are automatically considered to be failing. This can be confusing as functionality that is actually working, will be shown as failing.